### ISSUE TITLE
ANN version

### ISSUE THREAD

**Issue by naodaihuang** [cbergmeir/Rlgt#1]
Can you add a simpler ANN version of LGT?  In Smyl Slawek's 2015 paper Appendix 1, it has implementation of AAN, and "Remove or comment out highlighted lines for ANN model".   It would be similar to ses in forecast package.  

Will the ANN version work for time series that has negative values?

**Comment by slaweks17** [comment]
Hi,
Yes, it would be possible. I am actually planning to (re)implement a few ETS models in this framework, for the paper we are writing with Christoph Bergmeir. But does it make sense to expose it? I doubt it, more code to test, document etc. :-)
Negative values make you to use the additive seasonality, and this is more difficult to fit (it would probably end up "behind the scenes" as multiplicative, i.e. some seasonality component multiplied by level)
Cheers.

**Comment by naodaihuang** [comment]
It makes a lot of sense to have an ANN standalone function since it could be 10-100 times faster than current implementation?  Many other Bayesian methods were not well accepted in practice because of speed issues.  Only ANN, no others.  It's like in forecast package there's SES for ANN, and others are still in ETS.  There are many time series out there essentially ANN with no seasonality.  I guess then the input can take negative values assuming no seasonality?  Thanks for the great work by the way

**Comment by cbergmeir** [comment]
Hi,

well, the ses function in the forecast package is just a wrapper for the ets function, and my understanding is that it is mostly there for teaching purposes (i.e., the FPP book). But if it is really a lot faster I agree that it could make sense to have such a stand-alone function.

Regards,
Christoph

**Comment by naodaihuang** [comment]
yes ses is just a wrapper, but as I look at the stan code here, if it's ANN then fewer parameters and distributions needed, I guess it can be a lot faster

**Comment by slaweks17** [comment]
Christoph,
Unfortunately yesterday I did not manage to talk to my boss: I somehow forgotten about our meeting while watching “kids” (most Uber employees tend to be a bit younger than me 😊)  celebrating the float.
I will send her email first thing in the morning on Monday, and also try to grab her personally.

Regards,
Slawek

**Comment by slaweks17** [comment]
You are right, it is much faster. I implemented my version of damped Holt Winters and it is at least 10 times faster, and also, on M3 yearly dataset, beats ARIMA and ETS. 
So I warmed up to the idea of adding it, but then I tried to write a seasonal version and started having failures of Rstan (sampling does not work or kills R).  I have 2 days trying various combinations, nothing works. I asked a question on RStan forum - no answer so far.
But, this was all on Windows. I will try to do it on Linux.

**Comment by slaweks17** [comment]
Or wait till RStan 2.19 comes out.

**Comment by naodaihuang** [comment]
I'd suggest just a very simple one without seasonality consideration.  Many time series are not seasonal, and people can do their own seasonality trend decomposition anyway
